### PR TITLE
implement more numpy functions

### PR DESCRIPTION
**Pull request by ev-br**
work through the lists in gh-87.

- copyto
- resize
- histogram

### CODE REVIEW

File: torch_np/_funcs.py [hunk 1]
```diff
@@ -1048,6 +1058,26 @@ def tile(A: ArrayLike, reps):
...
+
+    numel = math.prod(new_shape)
+    quot, rem = divmod(numel, a.numel())
+    repeats = quot * a.numel()
+
+    result = torch.empty(numel, dtype=a.dtype)
+    result[:repeats] = torch.tile(a.ravel(), (quot,))
+    if rem > 0:
+        result[-rem:] = a.ravel()[:rem]
+    return result.reshape(new_shape)
```
[the remark this answers is not in the document]
Comment (ev-br) [reply torch_np/_funcs.py:1078]: yup, done in gh-106

File: torch_np/tests/numpy_tests/core/test_multiarray.py [hunk 2]
```diff
@@ -4549,6 +4550,7 @@ def test_none_shape(self):
         x.resize()
         assert_array_equal(x, np.eye(3))
 
+    @pytest.mark.xfail(reason="why would one do it")
```
[the remark this answers is not in the document]
Comment (ev-br) [reply torch_np/tests/numpy_tests/core/test_multiarray.py:4553]: Turns out `ndarray.resize` is documented to differ from `np.resize`, so many of these corner cases are in fact not corner cases but documented differences:

> If the new array is larger than the original array, then the new array is filled with repeated copies of a. Note that this behavior is different from a.resize(new_shape) which fills with zeros instead of repeated copies of a.

https://numpy.org/doc/stable/reference/generated/numpy.resize.html

One other annoyance is the support of both `resize(tuple)` and `resize(*tuple)`. Anyhow, this is all done in gh-106.

**Comment by ev-br** [comment]
Continued in gh-106 to pick up the refactor from gh-105.
